src/data_submission.js

```diff
@@ -1,67 +1,93 @@
-import { UPLOAD_URL, CODE_API_URL } from "../globals";
-
-// function to submit data that ends up in s3
-export async function submitData(dataset, id, token) {
-	const data = {
-		taskId: id,
-		dataset: dataset,
-	};
-	try {
-		const response = await fetch(UPLOAD_URL, {
-			method: "POST",
-			mode: "cors",
-			headers: {
-				"Content-Type": "application/json",
-				Authorization: token,
-			},
-			body: JSON.stringify(data),
-		});
-
-		const result = await response;
-		return result;
-	} catch (error) {
-		console.error("Data upload failed", error);
-	}
-}
+const API_URL = "https://mjbhgmtnxe.execute-api.eu-west-2.amazonaws.com/prod/KaptaLite_test";
+const BUCKET_BASE_URL = "https://s3.eu-west-2.amazonaws.com/kapta-lite-private-maps";
 
-// function to get task details from task dynamodb
-export async function getTaskDetails(code, token) {
-	const url = `${CODE_API_URL}/requests/code/${code}`;
-	try {
-		const response = await fetch(url, {
-			method: "GET",
-			mode: "cors",
-			headers: {
-				"Content-Type": "application/json",
-				Authorization: token,
-			},
-		});
-		const result = await response.json();
-
-		const info = JSON.parse(result); // it's returning the whole thing as a single object
-		return info;
-	} catch (error) {
-		console.error("Data retrieval failed", error);
-	}
-}
+export async function uploadProcessedChat(file, fileNameWAMap, setStatusText, setButtonDisabled, sharingOption, taskId, WhatsAppMapTags, wabMapperId) {
+    setStatusText("Preparing for sharing...");
+    setButtonDisabled(true);
+
+    try {
+        const visibility = sharingOption; // "private-sensitive", "private-non-sensitive", or "open"
+        const taskIdFolder = taskId || "noTaskId";; // To classify data by taskId, and give a value if no value
+        const tagsFolder = WhatsAppMapTags || "noMapTags"; // To classify data by tags
+        const WABMapperFolder = wabMapperId || "noWabMapperId"; // WhatsApp Mapper ID
+
+        console.log("📦 Sending to /download-url:", {
+            fileName: fileNameWAMap,
+            visibility,
+            taskIdFolder,
+            tagsFolder,
+            WABMapperFolder
+        });
+
+        // Step 1: Request a pre-signed URL from the backend
+        const response = await fetch(`${API_URL}/upload-url`, {
+            method: "POST",
+            body: JSON.stringify({
+                fileName: fileNameWAMap,
+                fileType: file.type,
+                visibility,
+                taskIdFolder,
+                tagsFolder,
+                WABMapperFolder
+            }),
+            headers: { "Content-Type": "application/json" }
+        });
+
+        if (!response.ok) throw new Error("Failed to get pre-signed URL");
```

Comment:
When dealing with promises you can catch errors directly rather than using a conditional to test for them